prefect_ray/task_runners.py

```diff
@@ -145,7 +145,9 @@ async def submit(
                 "The task runner must be started before submitting work."
             )
 
-        call_kwargs = self._exchange_prefect_for_ray_futures(call.keywords)
+        call_kwargs, upstream_ray_obj_refs = self._exchange_prefect_for_ray_futures(
```

Comment:
We may be able to just extract upstream ray object refs in this step then do the exchange with the resolved objects in `_run_prefect_task` — seems feasible to do as a follow-up though.

Reply:
true. however, we need to exchange the prefect futures with something else as they are not serializable and cannot be passed to a ray remote function. I think we would need in any case some sort of placeholder object which is then exchanged by the resolved object in _run_prefect_task. but a ray object ref is already a good candidate for such a placeholder and saves us from yet another indirection.